cli/azd/pkg/ext/models.go

```diff
@@ -63,6 +63,9 @@ type HookConfig struct {
 	Name string `yaml:",omitempty"`
 	// The type of script hook (bash or powershell)
 	Shell ShellType `yaml:"shell,omitempty"`
+	// If true, azd won't try to use alternative shell to execute the command. For example, pwsh won't be tried with
+	// powershell (pwsh5) in Windows.
+	StrictShell bool `yaml:"strictShell,omitempty"`
```

Comment:
This feels like a weird feature.  Instead of introducing a new config like "strictShell" and alternative design would be to have a scripts and which shell to execute them in.

Then in our implementation we could check if that shell is available.  If the shell is available invoke the script.  If not proceed down the list to the next shell and repeat.

This would actually make it easier so the configuration can be less OS specific especially for the cases when cross platform users have multiple shells available.

Reply:
> Then in our implementation we could check if that shell is available. If the shell is available invoke the script. If not proceed down the list to the next shell and repeat.

Do you mean, adding the list of shells to the hook configuration?  In other words, supporting an array-list for the Shell field?

I didn't want to go that route b/c we are trying to fix all current azd templates w/o asking all of them to update to a new Shell configuration for the hooks.

That's why the change from this PR makes powershell5 an alternative for pwsh7 automatically (no changes required in azure.yaml).

The strictShell option is a way to opt-out from the shell alternatives, resulting in the actual behavior.